r/R/dplyr-funcs-datetime.R

```diff
@@ -189,6 +189,70 @@ register_bindings_datetime <- function() {
   })
 }
 
+register_bindings_duration <- function() {
+  register_binding("difftime", function(time1,
+                                        time2,
+                                        tz,
+                                        units = c("auto", "secs", "mins",
+                                                  "hours", "days", "weeks")) {
+    units <- match.arg(units)
+    if (units != "secs") {
+      abort("`difftime()` with units other than seconds not supported in Arrow")
+    }
+
+    # for time32() we do not need to worry about timezone
+    if (call_binding("is.instant", time1) & call_binding("is.instant", time2)) {
```

Comment:
we might want look at where is.instant is defined and choose a different one?

Reply:
For some reason (I don't remember exactly why, but it's linked to the `time32()` comment) I wanted to split the logic into time types (`time32` & `time64`) vs date types (`date32` and `date64`) + timestamp. Hence the use of `is.instant`, which I think does what I was after, but not quite remember the _why_. At that moment I thought the comment was good enough.

Reply:
Ah ok, yeah date32, date64, and timestamp all have TZ attributes in Arrow, but time32 and time64 do not. 

> therefore we might want to recast to make sure

Let's construct (and maybe you have below and can point to them — but I didn't see something obvious that was this!) test cases where (re)casting them would change the answer. I suspect we might be slightly confusing the TZ attribute on the object and the TZ argument as an argument (or at least, I don't 100% see what cases would have issues that would need this recasting).

Reply:
See below, but I think it's safe to ignore the `tz` argument since it only produces a material impact for POSIXlt objects.